src/api/user/UserAPI.js

```diff
@@ -90,6 +91,50 @@ class UserAPI extends EventEmitter {
         }
     }
 
+    getPossibleRoles() {
+        if (!this.hasProvider()) {
+            return Promise.resolve(undefined);
+        } else {
+            return this._provider.getPossibleRoles();
+        }
+
+    }
+    /**
+     * If a user provider is set, it will return the active role Id
+     * @returns object
+     */
+    getActiveRole() {
+        if (!this.hasProvider()) {
+            return Promise.resolve(undefined);
+        }
+
+        // get from session storage
+        const sessionStorageValue = SessionPersistance.getActiveRole();
+        if (sessionStorageValue === 'undefined' || sessionStorageValue === undefined) {
+            return undefined;
+        }
+
+        return sessionStorageValue;
+    }
+    setActiveRole(role) {
+        SessionPersistance.setActiveRole(role);
+    }
+
+    /**
+     * Will return if a role can provide a operator status response
+     * @memberof module:openmct.UserApi#
+     * @returns {Boolean}
+    */
+    canProvideStatusForRole() {
```

Comment:
We should make this method async

Reply:
Just circling back on this, I'm not sure if the async declaration is needed for this particular function. As it stands currently, the sessionStorage portion and example user `canProvideStatusForRole` are both synchronous activities.